code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java

```diff
@@ -70,6 +72,7 @@ class AEPMessage implements FullscreenMessage {
     private Animation dismissAnimation;
     private Animation.AnimationListener animationListener;
     private Map<String, String> assetMap = Collections.emptyMap();
+    private final Executor executor;
```

Comment:
Can you avoid having an executor inside each AEPMessage instance? You can have a single lazy initialized executor in Android UI Service and pass it to the constructor.